recipes/libavif/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+import functools
+import os
+import textwrap
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+codecs = {
+    "aom": "libaom-av1/3.1.2",
+    "dav1d": "dav1d/0.9.1",
+}
+
+default_codec = "aom"
+
+
+class LibAVIFConan(ConanFile):
+    name = "libavif"
+    description = "Library for encoding and decoding .avif files"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/AOMediaCodec/libavif"
+    topics = "avif"
+    license = "BSD-2-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_codec": list(codecs.keys()),
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_codec": default_codec,
+    }
+    generators = "cmake", "cmake_find_package_multi"
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires(codecs[str(self.options.with_codec)])
+        # FIXME: require once libyuv is at least version 1774
+        # self.requires("libyuv/cci.20201106")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        root = self._source_subfolder
+        get_args = self.conan_data["sources"][self.version]
+        tools.get(**get_args, destination=root, strip_root=True)
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        suffix = str(self.options.with_codec).upper()
+        cmake.definitions[f"AVIF_CODEC_{suffix}"] = True
```

Comment:
For libaom, you can additionally pass -DAVIF_CODEC_AOM_DECODE=OFF to cmake so that libaom is only used as an AV1 encoder.

Reply:
I'm not sure that's an option worth exposing here. What is the value proposition of that?

Reply:
If -DAVIF_CODEC_AOM=ON is passed to cmake, libaom can be used as both the AV1 encoder and the AV1 decoder. If -DAVIF_CODEC_DAV1D=ON is also passed to cmake, both dav1d and libaom can be used as AV1 decoders, with dav1d preferred over libaom. It suffices to just have dav1d as the AV1 decoder. That's why I suggested passing -DAVIF_CODEC_AOM_DECODE=OFF to cmake.

The code disabled by -DAVIF_CODEC_AOM_DECODE=OFF is mainly in libavif/src/codec_aom.c, some in libavif/src/avif.c. You can evaluate it and see if it's worth making recipes/libavif/all/conanfile.py more complicated.

Reply:
So, the value proposition here is that when possible, libavif prefers dav1d, because it's better in some way? If so, I guess it's no biggie to create separate `with_encoder` and `with_decoder` options.

Reply:
Okay, I had wrong assumptions. I implemented `with_decoder` only instead with the encoder being `aom` always.

Reply:
@friendlyanon I somehow forgot to reply to your last question. Sorry. Yes, as an AV1 decoder, dav1d has much better performance than libaom.

Reply:
Yes, I assumed as such, that's why I set it as the default decoder. Thanks for confirming!